src/fu-plugin.c

```diff
@@ -906,7 +907,14 @@ fu_plugin_device_read_firmware (FuPlugin *self, FuDevice *device, GError **error
 		return FALSE;
 	if (!fu_device_detach (device, error))
 		return FALSE;
-	fw = fu_device_read_firmware (device, error);
+	firmware = fu_device_read_firmware (device, error);
+	if (firmware == NULL) {
+		g_autoptr(GError) error_local = NULL;
+		if (!fu_device_attach (device, &error_local))
+			g_debug ("ignoring: %s", error_local->message);
```

Comment:
I think it would be good to:
1. prefix `error` here now in the read and write firmware cases.  In a debugging context you'll see `ignoring: foo` but it will be non-obvious which call is being debugged.
2. potentially change that `ignoring: %s` string for both cases.

Reply:
Agreed, will fix both issues, thanks.